tests/std/tests/P1222R4_flat_set/test.cpp

```diff
@@ -288,6 +299,33 @@ void test_extract() {
     assert(ranges::equal(cont, elements));
 }
 
+// TRANSITION, too simple
+void test_erase_1() {
+    flat_set<int> fs{1};
+    fs.erase(1);
+    assert_all_requirements_and_equals(fs, {});
+}
+
+template <class T>
+struct holder {
+    T t;
+    operator T() && {
+        return std::move(t);
+    }
+};
+
+void test_erase_2() {
+    using C = flat_set<int, std::less<>>;
+    C fs{1, 2, 3};
+    assert_all_requirements_and_equals(fs, {1, 2, 3});
+    // this should be allowed per P2077R3:
+    fs.erase(holder<C::iterator>{fs.begin()});

```

Comment:
This is the case demonstrated in [`Why K&& rather than const K&`](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2077r3.html#whyfwd)